### PR TITLE
[IMP] spreadsheet: allow to update pivot dimensions

### PR DESCRIPTION
**Pull request by LucasLefevre**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by rrahir** [comment]
@robodoo rebase-ff

File: addons/spreadsheet/static/src/o_spreadsheet/o_spreadsheet.xml [hunk 1]
```diff
@@ -2,8 +2,8 @@
  This file is generated by o-spreadsheet build tools. Do not edit it.
  @see https://github.com/odoo/o-spreadsheet
  @version 17.2.0-alpha.6
- @date 2024-02-28T13:13:43.538Z
- @hash 318a04ee6
```
Comment (rrahir) [inline addons/spreadsheet/static/src/o_spreadsheet/o_spreadsheet.xml:6]: to remove

File: addons/spreadsheet/static/src/o_spreadsheet/o_spreadsheet.js [hunk 2]
```diff
@@ -59788,8 +59788,8 @@ stores.inject(MyMetaStore, storeInstance);
 
 
     __info__.version = "17.2.0-alpha.6";
-    __info__.date = "2024-02-28T13:13:41.220Z";
```
Comment (rrahir) [inline addons/spreadsheet/static/src/o_spreadsheet/o_spreadsheet.js:59791]: to remove

File: addons/spreadsheet/static/src/pivot/plugins/pivot_core_plugin.js [hunk 3]
```diff
@@ -115,10 +115,15 @@ export class PivotCorePlugin extends OdooCorePlugin {
                 this.history.update("nextFormulaId", parseInt(newPivotId, 10) + 1);
                 break;
             }
+            // this command is deprecated. use UPDATE_PIVOT instead
```
Comment (rrahir) [inline addons/spreadsheet/static/src/pivot/plugins/pivot_core_plugin.js:118]: if we deprecate it, I think it'd be nice to change the tests accordingly.

File: addons/web/static/tests/legacy/helpers/utils.js [hunk 4]
```diff
@@ -993,7 +993,7 @@ export async function drag(from, pointerType = "mouse") {
         targetPosition = getTargetPosition(position);
 
         // Move, enter and drop the element on the target
-        await triggerEvent(source, null, "pointermove", targetPosition);
+        await triggerEvent(source, null, "pointermove", { ...targetPosition, button: -1 });
```
[the remark this answers is not in the document]
Comment (LucasLefevre) [reply addons/web/static/tests/legacy/helpers/utils.js:996]: Yes it is :/

[the remark this answers is not in the document]
Comment (LucasLefevre) [reply addons/web/static/tests/legacy/helpers/utils.js:996]: from https://github.com/odoo/o-spreadsheet/pull/3721

> Note that there's a difference for the button property between
mousemove and pointermove.
For pointermove: "the button property has the value -1 when the button state has not changed from the previous pointermove event".
See https://github.com/w3c/pointerevents/issues/33

**Review by pro-odoo — APPROVED** [review]
robodoo rebase-ff r+

**Pull request closed without merging**